[18.0][MIG] product_pricelist_fixed_currency_rate: Migration to 18.0

BhaveshHeliconia opened the pull request.

LoisRForgeFlow committed:
[ADD] product_pricelist_fixed_currency_rate

Commit:
[UPD] Update product_pricelist_fixed_currency_rate.pot

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_fixed_currency_rate
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_fixed_currency_rate/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (12 of 12 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_pricelist_fixed_currency_rate
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_pricelist_fixed_currency_rate/es/

kobros-tech committed:
[MIG] product_pricelist_fixed_currency_rate: Migration to 17.0

Commit:
[UPD] Update product_pricelist_fixed_currency_rate.pot

OCA-git-bot committed:
[BOT] post-merge updates

BhaveshHeliconia committed:
[MIG] product_pricelist_fixed_currency_rate: Migration to 18.0

BhaveshHeliconia commented:
/ocabot migration product_pricelist_fixed_currency_rate

BhaveshHeliconia commented:
Hi @LoisRForgeFlow,

Could you please review this pull request when you have a moment? I would appreciate any feedback or suggestions for improvement.

Thank you for your time!

LoisRForgeFlow commented:
Hi @HeliconiaSolutions 

Thanks for your contribution, before reviewing I would like that CI runs its checks, I cannot accept the workflow so we need to ask @OCA/product-maintainers to do so.

At the same time, I'm not sure what is the OCA policy regarding contributions from a company account, not sure if it is required that contributions come from individual accounts. @pedrobaeza @etobella maybe one of you can clarify this?

Kind regards,

pedrobaeza commented:
There's no strict prohibition of these accounts, but first of all, the CLA should be signed, and second, I don't like these techniques, but it's just a personal POV.

rousseldenis commented:
@LoisRForgeFlow Already stated about that on other PR's. Doing a PR with a global account is ok IMHO. But for commits, I tend to say they should have beeen done with an individual account.

@pedrobaeza The CLA bot was looking at commiters logins. And as only individuals are mentioned in CLAS (even there is a company one), the bot was blocking such PR's.

pedrobaeza commented:
Well, apart from not seeing the CLA bot warning since a while, I think it's incorrect to not warn in these cases. A possible workaround is to put this login in a child account or similar to act like an individual.

LoisRForgeFlow commented:
Yes, it is also a while since I see one of those CLA warnings

@pedrobaeza I also find it a bit weird to contribute with a company account. I see problems with responsibility and attribution; especially internally it must generate quite some headaches when the company have quite a few developers/contributors. 

> @LoisRForgeFlow Already stated about that on other PR's. Doing a PR with a global account is ok IMHO. But for commits, I tend to say they should have beeen done with an individual account.

This can be a good intermediate approach, to require that at least commits come from an individual.

pedrobaeza commented:
You have GitHub organizations for pushing the branches, so there's no sense in pushing under a company account.

LoisRForgeFlow commented:
Then, we are in a all-or-nothing decision and from what you said before, this has been already discussed and the decision (maybe with no full agreement) was that it is allowed to collaborate with company accounts. Am I getting it right?

BhaveshHeliconia commented:
> There's no strict prohibition of these accounts, but first of all, the CLA should be signed, and second, I don't like these techniques, but it's just a personal POV.

@pedrobaeza, we’ve signed the entity CLA, which might be why you haven’t seen the CLA warning from the OCA bot.

BhaveshHeliconia commented:
> Yes, it is also a while since I see one of those CLA warnings
> 
> @pedrobaeza I also find it a bit weird to contribute with a company account. I see problems with responsibility and attribution; especially internally it must generate quite some headaches when the company have quite a few developers/contributors.
> 
> > @LoisRForgeFlow Already stated about that on other PR's. Doing a PR with a global account is ok IMHO. But for commits, I tend to say they should have beeen done with an individual account.
> 
> This can be a good intermediate approach, to require that at least commits come from an individual.

@LoisRForgeFlow @pedrobaeza 
Thanks for sharing your thoughts, and we completely understand your concerns. We decided to adopt this approach because we've observed several companies doing the same, and it helps centralize contributions in a more unified manner. By using the company account, we aim to streamline the process and ensure that the entire organization shares responsibility and attribution for the work done.

That said, we're fully open to both individual and company-based contributions. We believe each approach has its merits, and we're happy to adapt based on what works best for the team and the project.

pedrobaeza commented:
Using the organization GitHub remote, any in the company can push to that branch, so about the flows, you won't have any restriction.

BhaveshHeliconia commented:
@pedrobaeza @LoisRForgeFlow @rousseldenis Thanks for the sharing the information.

As per our recent discussions, we have implemented the necessary changes on our side and have also signed the updated Individual CLA. Additionally, we’ve updated our profile to reflect these changes. To further support the OCA community, we’ve become proud members of OCA for 2025.

We’re excited about the opportunities for collaboration and look forward to working together with the entire OCA community to build even better open-source solutions.

LoisRForgeFlow commented:
Hi @HeliconiaSolutions

Thanks for that. 

I would kindly as again @OCA/product-maintainers to accept the CI workflow of this PR, which is the first automatic review that is needed in order to move forward with this pull request.

Thanks!

sebalix commented on product_pricelist_fixed_currency_rate/tests/test_product_pricelist_fixed_currency_rate.py:
```diff
@@ -12,85 +12,97 @@ def setUpClass(cls):
-                "name": "Test",
-                "currency_id": cls.env.ref("base.EUR").id,
-            }
+            [
+                {
+                    "name": "Test",
+                    "currency_id": cls.env.ref("base.EUR").id,
+                }
+            ]
```
IMO this change (in the whole test file) is not required (`create` accepts a dictionary), and introduces diff that could be annoying when porting commits.

sebalix reviewed (commented):
LG, minor comment

BhaveshHeliconia commented:
Hello @sebalix,

Thanks for the feedback! I see your point that the change might not be necessary, especially since create already accepts a dictionary. I have reverted the changes to avoid adding unnecessary diffs and potential conflicts when porting commits.

LoisRForgeFlow reviewed (approved):
Functional and code review LGTM

LoisRForgeFlow commented:
/ocabot merge nobump
